backend/data/utility.py

```diff
@@ -21,3 +21,52 @@ def create_delta_cols( df,group_by_cols, delta_needed_for_cols):
     df = df.set_index(index_cols)
 
     return df
+
+
+def interpolate_values(df, group_by_cols, delta_needed_for_cols):
+    # create copy of original dataframe
+    df_cp = df.copy()
+
+    # concatenate join columns
+    df_cp["combo"] = df_cp[group_by_cols].apply(lambda x: "|".join(x.tolist()), axis=1)
```

Comment:
this can be done by groupby or set_index on those group columns... merging them in to a 'key' string is a bit of a hack :-)

Reply:
Changed to groupby